### PR TITLE
Update for new Argonaut major version

### PR DESCRIPTION
**Pull request by thomashoneyman**
Argonaut's type-class-based codecs [now use typed errors instead of string errors](https://github.com/purescript-contrib/purescript-argonaut-codecs/releases/tag/v7.0.0), so I'm updating affected libraries for compatibility with the new version.

I've bumped dependency versions in the Bower file, but the only major version bump is the `argonaut-codecs` version. Fortunately, this library compiles with both v6.0.0 and v7.0.0 (the new version), so even if Hyper isn't updated when the new package set comes out it will still build with the package set. I also removed the unnecessary dependency resolution for typelevel-eval.

While here, I noticed that the spago.dhall file is using the old format -- `mkPackage` doesn't exist anymore -- so I updated it to use a more recent package set and removed that line. I also noticed that the version of Pulp used in the package.json isn't compatible with the latest PureScript (v0.13.8), so I've updated it.

While verifying that the project still compiles and passes tests with my changes I noticed that the tests fail to compile on v0.13.8 of the compiler due to a missing type annotation. I've added that type annotation so the tests work properly. I also noticed some compiler warnings about duplicate imports, which I've removed.

tldr; dependencies are updated, this library is compatible with the upcoming package set, and it is now also compatible with the latest PureScript tooling.

### CODE REVIEW

**Comment by thomashoneyman** [comment]
Quick update: most of the other affected libraries have been updated, so we'll cut a new package set soon. If it's possible to cut a new release of this library with the change sometime this week that would help ensure the transition is smooth. Let me know if there's anything else I can do to make sure this looks good!

**Comment by thomashoneyman** [comment]
Thanks! If you're also able to make a release then I can use that release to update the Bower files of [trout-client](https://github.com/purescript-hyper/purescript-trout-client) and [hypertrout](https://github.com/purescript-hyper/purescript-hypertrout). It's not necessary wrt package sets, but that way Bower users are still covered.

**Comment by nsaunders** [comment]
No problem @thomashoneyman and sorry for the delay. The new tag is `v0.11.1`. :-)

**Comment by thomashoneyman** [comment]
Thanks so much!
